### PR TITLE
Allow php8 & sylius 1.10

### PR DESCRIPTION
**Pull request by TonySma**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -31,13 +31,12 @@
         "mockery/mockery": "^1.4",
         "pamil/prophecy-common": "^0.1",
         "phpspec/phpspec": "^6.1",
-        "phpstan/phpstan": "0.12.29",
-        "phpstan/phpstan-doctrine": "0.12.16",
+        "phpstan/phpstan": "^0.12.57",
+        "phpstan/phpstan-doctrine": "^0.12.19",
```
Comment (maximehuran) [inline composer.json:35]: Are you sure that we break nothing with that ?

File: src/DependencyInjection/UiElementRegistryPass.php [hunk 2]
```diff
@@ -75,7 +75,9 @@ public function process(ContainerBuilder $container): void
      */
     private function validateUiElementResource(string $class): void
     {
-        if (!\in_array(UiElementInterface::class, class_implements($class), true)) {
+        $interfaces = (array) (class_implements($class) ?? []);
```
Comment (maximehuran) [inline src/DependencyInjection/UiElementRegistryPass.php:78]: Why this change ?
It seems that this methods always returns an array : https://www.php.net/manual/fr/function.class-implements.php

File: composer.json (hunk 1 above)
Comment (TonySma) [reply composer.json:35]: it's the minimal versions that allow php8
Sylius is even highter: 
https://github.com/Sylius/Sylius/blob/c2d97c8dd44530d59d588f41498042950f801c20/composer.json#L188

File: src/DependencyInjection/UiElementRegistryPass.php (hunk 2 above)
Comment (TonySma) [reply src/DependencyInjection/UiElementRegistryPass.php:78]: https://www.php.net/manual/fr/function.class-implements.php#refsect1-function.class-implements-returnvalues
says "Retourne un tableau ou false en cas d'erreur."

also in phpdoc: 
```
/**
 * Return the interfaces which are implemented by the given class
 * @link https://php.net/manual/en/function.class-implements.php
 * @param object|string $object_or_class <p>
 * An object (class instance) or a string (class name).
 * </p>
 * @param bool $autoload [optional] <p>
 * Whether to allow this function to load the class automatically through
 * the __autoload magic
 * method.
 * </p>
 * @return string[]|false An array on success, or false on error.
 */
 ```
the in_array require an array as second parameter: 
https://www.php.net/manual/fr/function.in-array.php

and phpstan trigger an error without this
```
vendor/bin/phpstan analyse -c phpstan.neon src                                                                                                                                                      git:(sylius1.10|) 
 32/32 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ ----------------------------------------------------------------------------------------------- 
  Line   DependencyInjection/UiElementRegistryPass.php                                                  
 ------ ----------------------------------------------------------------------------------------------- 
  80     Parameter #2 $haystack of function in_array expects array, array<string, string>|false given.  
 ------ ----------------------------------------------------------------------------------------------- 

                                                                                                                        
 [ERROR] Found 1 error                                                                                                  
```

**Review by maximehuran — APPROVED** [review]

**Review by jacquesbh — APPROVED** [review]

**Comment by jacquesbh** [comment]
Thank you @TonySma ! 🎉
